### ISSUE TITLE
Allow image uploads with pre-signed url

### ISSUE THREAD

**Issue by sandip-mane** [neetozone/neeto-editor#308]
The image uploads need to work with a pre-signed URL
To make it happen, the editor can expect direct uploads URL.

```js
<Editor 
  directUploadsURL={`api/v1/direct_uploads`} 
/>
```

This direct upload URL will give a pre-signed URL to which the editor can upload the actual file.
To see how direct uploads work, refer following:

DirectUploads definition in Rails
https://github.com/rails/rails/blob/aaa64687e81873b26c964fada0b969621ae8cf8f/activestorage/app/javascript/activestorage/direct_upload.js#L7

How that's used: 

https://github.com/bigbinary/neeto-planner-web/blob/1d188604cf5d77f43dbb91f3cf38cb9f129d67e1/app/javascript/src/components/Layouts/Header/Attachment.jsx#L36-L46

**Comment by karthiknmenon** [comment]
@AbhayVAshokan 

Let's create a custom hook `useDirectUpload` that takes in a URL and returns  `{upload, cancel, progress, error, isUploading, previewUrl, meta}`. 

- upload - is used to upload a given file to the `direct_upload` endpoint.
- cancel - can be used to cancel a given upload anytime. 
- progress - returns the file upload progress, which can be used to show a progress bar in the UI.
- error - returns errors, if any, during the upload.
- isUploading - a state that can be used to disable few CTAs in the host application  (if needed)
- previewUrl - can be used to have a preview of the uploaded image/file.
- meta - includes things like extension, size, filename etc.

A similar hook has been added [here](https://github.com/bigbinary/neeto-crm-web/pull/1040). We can build one here and move it to neeto-commons. 

@Amaljith-K please lmk your thoughts on this. 

I am not sure if we are allowing multiple uploads in neeto products right now. If that is the case, then we can handle the case of enquiring and dequeuing uploads too. But I don't think it's needed at this moment here. 

cc: @labeebklatif

**Comment by AbhayVAshokan** [comment]
@karthiknmenon 

The issue https://github.com/bigbinary/neeto-planner-web/issues/1563 and the associated roadmap have a similar discussion. I recall that @Melvin9 would be working on this issue.

TLDR; file upload logic will be extracted to an NPM package cum gem approach and will be used across all neeto projects.

As you mentioned above, I shall make a reusable utility function to handle the upload logic. This can be extracted to the proper repository later on.

**Comment by neerajsingh0101** [comment]
I think we had decided to use
https://www.npmjs.com/package/uppy

**Comment by AbhayVAshokan** [comment]
@neerajdotname we are using `uppy` for neetoEditor image uploads. It is not as extensive as discussed in the planner roadmap (a dashboard with multiple upload options). I shall add the direct-upload logic to the existing `uppy` uploader implementation so that it would be easier to build the new package.
